### DISCUSSION TITLE
Why TS types of Plugins written in TS are not emitted and exposed?

### DISCUSSION THREAD

**Discussion by MananTank** [babel/babel#14738] [General] [answered]
For Example: [babel-helper-module-imports](https://github.com/babel/babel/tree/main/packages/babel-helper-module-imports) is written in TS

But I don't see their types being exposed in package.json > types

Why is that? Is this work in progress?

**Comment by nicolo-ribaudo** [comment, marked as answer]
We recently (almost) finished migrating our repository to TS, and the types still don't have a good quality. Additionally, many of the new type definitions are incompatible with what's currently on DefinitelyTyped. We will probably wait at least for Babel 8 before publishing them.

**Comment by MananTank** [comment]
Got it, Thanks 🙌
